Content.Server/Doors/Electronics/Systems/DoorElectronicsSystem.cs

```diff
@@ -0,0 +1,84 @@
+using Content.Shared.Doors;
+using Content.Shared.Doors.Electronics;
+using Content.Shared.Interaction;
+using Content.Server.Doors.Electronics;
+using Robust.Server.GameObjects;
+
+namespace Content.Server.Doors.Electronics;
+
+public sealed class DoorElectronicsSystem : EntitySystem
+{
+    [Dependency] private readonly UserInterfaceSystem _uiSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<DoorElectronicsComponent, DoorElectronicsUpdateConfigurationMessage>(OnChangeConfiguration);
+        SubscribeLocalEvent<DoorElectronicsComponent, DoorElectronicsRefreshUiMessage>(OnRefreshUi);
+        SubscribeLocalEvent<DoorElectronicsComponent, InteractUsingEvent>(OnInteractUsing);
+    }
+
+    public void UpdateUserInterface(EntityUid uid, DoorElectronicsComponent component)
+    {
+        List<string> accesses = new List<string>();
+
+        if (TryComp<AccessReaderComponent>(component.Owner, out var accessReader))
+        {
+            foreach (var accessList in accessReader.AccessLists)
+            {
+                var access = accessList.FirstOrDefault();
+                if (access == null)
+                    continue;
+                accesses.Add(access);
+            }
+        }
+
+        var state = new DoorElectronicsConfigurationState(accesses);
+
+        _uiSystem.TrySetUiState(uid, DoorElectronicsConfigurationUiKey.Key, state);
+    }
+
+    private void OnChangeConfiguration(EntityUid uid,
+                                       DoorElectronicsComponent component,
+                                       DoorElectronicsUpdateConfigurationMessage args)
+    {
+        if (TryComp<AccessReaderComponent>(uid, out var accessReader))
+        {
+            accessReader.AccessLists.Clear();
+            foreach (var access in args.accessList)
+            {
+                accessReader.AccessLists.Add(new HashSet<string>(){access});
+            }
+        }
+        var state = new DoorElectronicsConfigurationState(args.accessList);
+        _uiSystem.TrySetUiState(component.Owner,
+                                DoorElectronicsConfigurationUiKey.Key,
+                                state);
+    }
+
+    private void OnRefreshUi(EntityUid uid,
+                             DoorElectronicsComponent component,
+                             DoorElectronicsRefreshUiMessage args)
+    {
+        UpdateUserInterface(uid, component);
+    }
+
+    private void OnInteractUsing(EntityUid uid, DoorElectronicsComponent component, InteractUsingEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (!TryComp<NetworkConfiguratorComponent>(args.Used, out var networkConfigurator))
+            return;
+
+        if (!TryComp(args.User, out ActorComponent? actor))
+            return;
+
+        args.Handled = true;
+
+        _uiSystem.TryOpen(uid, DoorElectronicsConfigurationUiKey.Key, actor.PlayerSession);
```

Comment:
This review is not a requirement but a recommendation, so you decide whether or not you want to do this. Replace this with an `ActivatableUIComponent`, tho it needs to support interacting *with* something to open the UI, this should be easy to do as i *think* you only need to add an `EntityWhitelist?` datafield to the component and then handle `InteractUsingEvent` on the `ActivatableUISystem` (filtering the used entity through `whitelistField.IsValid(args.Used, EntityManager)` method first, of course), and finally set the whitelist datafield to NetworkConfigurator in the prototype, feel free to ask any question.

Reply:
I tried, but I get a serialization error and I don't know how to fix it (or what it even means).
`[ERRO] proto: Reading Robust.Shared.Prototypes.EntityPrototype(DoorElectronicsArmory) threw the following exception: Robust.Shared.Serialization.Manager.Exceptions.RequiredFieldNotMappedException: Required field key of type System.String wasn't mapped.`

Reply:
push the code with that error and i will investigate

Reply:
Here, it's done.